cmd/entrypoint/runner.go

```diff
@@ -33,7 +35,25 @@ func (rr *realRunner) Run(args ...string) error {
 	signal.Notify(rr.signals)
 	defer signal.Reset()
 
-	cmd := exec.Command(name, args...)
+	// Add timeout to context if a non-zero timeout is specified for a step
+	ctx := context.Background()
+	var cancel context.CancelFunc
+
+	deadline, _ := time.Parse(time.UnixDate, taskRunDeadline)
+	taskRunDuration := time.Until(deadline)
+
+	stepDuration, _ := time.ParseDuration(timeout)
+	timeoutDuration := stepDuration
+
+	if timeoutDuration != time.Duration(0) && taskRunDeadline != "" && deadline.Before(time.Now().Add(timeoutDuration)) {
```

Comment:
Does this not mean that the `taskRunDeadline` will only be applied if a Step timeout has also been set?  What happens if `timeout` is "0s" or ""?

A helper func might be useful here:

```go
// soonestTimeout accepts a step and taskrun duration and returns whichever one is non-zero (if any)
// or that which will occur soonest.
func soonestTimeout(step, taskRun time.Duration) time.Duration {
	if step == time.Duration(0) {
		return taskRun
	}
	if taskRun == time.Duration(0) {
		return step
	}
	if step < taskRun {
		return step
	}
	return taskRun
}
```

Reply:
Oops messed up the commit version. Moved them to a separate function